tools/NodeConfigCompare/configuration/cloud_formation_template.yaml

```diff
@@ -389,18 +445,25 @@ Resources:
               - !Sub "arn:aws:redshift:${AWS::Region}:${AWS::AccountId}:parametergroup:${ClusterIdentifierPrefix}*"
           - Effect: Allow
             Action:
-              - ec2:Describe*
               - redshift:restoreFromClusterSnapshot
               - redshift:describeClusterSnapshots
-              - redshift-data:ExecuteStatement
-              - redshift-data:ListStatements
-              - redshift-data:GetStatementResult
-              - redshift-data:DescribeStatement
               - redshift:DescribeLoggingStatus
               - redshift:DescribeClusters
               - redshift:DescribeClusterParameters
             Resource:
-              - '*'
+              - !Sub "arn:aws:redshift:${AWS::Region}:${AWS::AccountId}:cluster:*"
+              - !Sub "arn:aws:redshift:${AWS::Region}:${AWS::AccountId}:snapshot:*"
+              - !Sub "arn:aws:redshift:${AWS::Region}:${AWS::AccountId}:dbname:*"
+              - !Sub "arn:aws:redshift:${AWS::Region}:${AWS::AccountId}:dbuser:*"
+              - !Sub "arn:aws:redshift:${AWS::Region}:${AWS::AccountId}:parametergroup:*"
+          - Effect: Allow
+            Action:
+              - ec2:Describe*
+              - redshift-data:ExecuteStatement
+              - redshift-data:ListStatements
+              - redshift-data:GetStatementResult
+              - redshift-data:DescribeStatement
+            Resource: "*"
```

Comment:
Lmk if this is not possible - why not split this into two different policies and specify the EC2 instance ARN and the cluster ARN (I believe the cluster one can be copied from the first IAM policy definition)

Reply:
For redshift-data, I tried adding the cluster ARNs. It did not work. It is probably because redshift-data does not support ARNs, so had to give `"*"`.
For EC2, I was looking at the actions the `Describe*` has a a lot of actions that does not require an ARN. 

But I can try splitting redshift-data and EC2, give `"*"` to redshift-data and add ARNs to EC2 and try if it works.